common/src/main/java/com/microsoft/identity/common/internal/dto/AccessToken.java

```diff
@@ -241,4 +245,16 @@ public String getExpiresOn() {
     public void setExpiresOn(final String expiresOn) {
         mExpiresOn = expiresOn;
     }
+
+    @Override
+    public boolean isExpired() {
+        // Init a Calendar for the current time/date
+        final Calendar calendar = Calendar.getInstance();
+        calendar.add(Calendar.SECOND, AuthenticationSettings.INSTANCE.getExpirationBuffer());
+        final Date validity = calendar.getTime();
+        // Init a Date for the accessToken's expiry
+        long epoch = Long.valueOf(getExpiresOn());
+        final Date expiresOn = new Date(epoch * 1000);
+        return expiresOn.before(validity);
```

Comment:
we need 5 minutes for clock skew, is that handled somewhere else?

Reply:
Line 253

Reply:
oops